Add auto-mapping for CSV column headers on upload

AgbediaSamuel opened the pull request:
Implements automatic mapping of CSV headers to fields by calling the backend API after file upload. If auto-mapping fails, users can still manually map columns. The parseCsvFile function is now async and uses the token for authorization.

AgbediaSamuel committed:
Add auto-mapping for CSV column headers on upload

Implements automatic mapping of CSV headers to fields by calling the backend API after file upload. If auto-mapping fails, users can still manually map columns. The parseCsvFile function is now async and uses the token for authorization.

AgbediaSamuel committed:
Refactor lead file upload to use local column auto-mapping

Replaces the API-based column auto-mapping in uploadLeadFile.tsx with a new local utility, autoMapColumns, implemented in src/utils/columnMapper.ts. This improves performance and reliability by removing the network dependency and provides a more maintainable mapping logic.

AgbediaSamuel committed:
Enhance upload pages with filters, sorting, and UI improvements

Added date range and sorting filters to the upload history page, improved layout and spacing for supervisor pages, and refactored the upload lead file page to show a mapping summary in the left column. Also improved label consistency and removed redundant code in columnMapper (one line).

IamWafula commented on src/pages/supervisorPages/showUploads.tsx:
```diff
@@ -524,8 +527,34 @@ const ShowUploads = () => {
+        end.setHours(23, 59, 59, 999); // End of day
+        return uploadDate <= end;
+      });
+    }
```
The date filter should be inclusive. 24 - 25 should include records on the 24th and 25th

AgbediaSamuel committed:
Squashed commit of the following:

commit fce1be093595494566764a2e0669b1766a5ac2fb
Author: Ian Wafula <46840113+IamWafula@users.noreply.github.com>
Date:   Tue Nov 25 15:21:11 2025 -0800

    Added Click ID

AgbediaSamuel commented:
This PR also contains small UI changes, didn't want to make a new PR for that

IamWafula commented:
Trial comment

AgbediaSamuel committed:
Refactor sorting UI and date filtering in ShowUploads

Replaces sorting buttons with a checkbox to toggle sorting by oldest uploads. Updates date filtering logic to compare date strings in YYYY-MM-DD format for more accurate filtering.

IamWafula commented:
LGTM
